### ISSUE TITLE
cdist NAN in backward

### ISSUE THREAD

**Issue by klicperajo** [pytorch/pytorch#37237]
## 🐛 Bug

In PyTorch 1.5 (not 1.4) using cdist produces NANs instead of regular gradients. This happens on both CPU and GPU.

## To Reproduce

1. [Download the tensors](https://figshare.com/s/47d6035b990fa62d730d)
2. Run the code
```python
import torch

emb1, emb2, cdist_grad = torch.load('cdist_grad.pt')

emb1.retain_grad()
d = torch.cdist(emb1, emb2)
d.backward(cdist_grad)

print(emb1.grad[0, 17])
```
3. The printed gradients will be NAN.

## Expected behavior

Actual numerical gradients.

## Environment

 - PyTorch Version (e.g., 1.0): 1.5.0
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.7.6
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: Nvidia 1080 Ti (but it also happens on CPU)
 - Any other relevant information:

**Comment by klicperajo** [comment]
Sorry, this seems to be a duplicate of #37154.

**Comment by gchanan** [comment]
Thanks for the report, let's keep the discussion in #37154
